src/app/pages/send-tokens/components/asset-search/asset-search.tsx

```diff
@@ -1,176 +1,56 @@
-
-    return (
-      <Fade in={isOpen && !!items?.length}>
-        {styles => (
-          <Stack
-            flexDirection="column"
-            boxShadow="0px 8px 16px rgba(27, 39, 51, 0.08);"
-            borderRadius="6px"
-            position="absolute"
-            width="100%"
-            top="77px"
-            maxHeight="230px"
-            border={isOpen ? '1px solid #E1E3E8' : 'none'}
-            zIndex={1000}
-            overflow="auto"
-            style={styles}
-            spacing="extra-loose"
-            bg={color('bg')}
-            p="loose"
-            ref={ref}
-            {...props}
-          >
-            {items?.map((asset, index) => {
-              const isStx = asset.type === 'stx';
-              if (isStx && !availableStxBalance) return null;
-              return (
-                <AssetRow
-                  isPressable
-                  asset={asset}
-                  index={index}
-                  key={`${asset.contractAddress || asset.name}__${index}`}
-                  data-asset={isStx ? 'stx' : asset.meta?.symbol}
-                  highlighted={highlightedIndex === index ? 'ink.150' : 'white'}
-                  {...getItemProps({ item: asset, index })}
-                />
-              );
-            })}
-          </Stack>
-        )}
-      </Fade>
-    );
-  }
-);
-const AssetSearchField = memo(
-  ({ autoFocus, onItemClick, ...rest }: { autoFocus?: boolean; onItemClick: () => void }) => {
-    const assets = useTransferableAssets();
-
-    const { selectedAsset, handleUpdateSelectedAsset } = useSelectedAsset();
-
-    const searchInput = useSearchInput();
-    const setSearchInput = useUpdateSearchInput();
+    const { selectedAsset, updateSelectedAsset } = useSelectedAsset();
+    const [searchInput, setSearchInput] = useState<string>('');
+    const [assetItems, setAssetItems] = useState(assets);
 
     useEffect(() => {
       if (principalHasOnlyOneAsset(assets ?? [])) {
-        handleUpdateSelectedAsset(assets[0]);
+        updateSelectedAsset(assets[0]);
       }
-    }, [assets, handleUpdateSelectedAsset]);
-
-    const {
-      isOpen,
-      getLabelProps,
-      getMenuProps,
-      getInputProps,
-      getComboboxProps,
-      highlightedIndex,
-      getItemProps,
-      openMenu,
-    } = useCombobox({
-      items: assets || [],
-      initialIsOpen: true,
-      inputValue: searchInput,
-      defaultIsOpen: false,
-      selectedItem: selectedAsset as any,
-      itemToString: item => {
-        return item?.contractAddress || item?.name || '';
-      },
-      onSelectedItemChange: ({ selectedItem }) => {
-        onItemClick();
-        handleUpdateSelectedAsset(selectedItem || undefined);
-      },
-    });
-
-    const labelRef = useRef(null);
-    const comboRef = useRef(null);
-
-    if (!assets) return null;
+      return () => onClearSearch();
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    const onClearSearch = () => {
```

Comment:
@kyranjamie I did more work to separate out these components and clean things up. What is your opinion on these functions here, do they need `useCallback`? I feel like we over use useCallback?  Was referencing this: https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
Thanks for sharing. Indeed, in components it's not so helpful, I've probably overused myself. 

Though I would push referential equality point, and go as far as saying that inside hooks we should _always_ use `useCallback` or `useMemo`. You never know when/where it'll be consumed, and new references on every render can cause hard to debug infinite render loops that freeze the app.

Reply:
Thanks for feedback. 👍